markdown/core.py

```diff
@@ -85,7 +85,7 @@ class Markdown:
     callable which accepts an [`Element`][xml.etree.ElementTree.Element] and returns a `str`.
     """
 
-    def __init__(self, **kwargs):
+    def __init__(self, **kwargs) -> None:
```

Comment:
This addition adds no value and I will not tolerate being forced to add it.

Reply:
Adding any annotation indicates "please type-check the body of this function". The cool thing about it is that if you add a new function and not write any type annotations on it, mypy will not complain whatsoever. It is a **good** thing to have this tool. So this un-nuanced take on this is unproductive.

Here I'm adding this `-> None` because this function contains some type annotations that are being wasted otherwise. To not have problems with this, I'll have to move the annotations to class level. Let me know if that works better for your preference.

Reply:
But when does an `__init__`  method return anything other than `None`? That's my point. Adding this annotation adds no value in that it tells me nothing I didn't already know. I should not be  required to add this annotation every time I define and `__init__` function.

You seem to object based on the fact that eliminating the annotation causes the tool to not run other checks on the line. That is a problem with the tool. If the tool can't do that, then fix the tool or use a different tool. Don't force me to waste time and effort adding otherwise pointless annotations.

Reply:
Adding `-> None` annotations to `__init__` **is** useless, I totally agree on that. However you ignore the point that I am actually arguing for. **You** want**he tool to not do anything unless you call for it. As such, not adding any annotations accomplishes that. But when you **do** want to go ahead and tell the tool "please type-check this function body", this is the designated way to do that. You are not forced, and I demonstrated that with the current state of this pull request.

Separately on your last point: the tool **does not** require this annotation, you can simply configure it to check all function bodies. If you want it to check only **some** function bodies, this is how to indicate it.